### PR TITLE
feat(catalog): implemented storybook

### LINKED ISSUE #6: Create components catalog
We should create a component catalog as the index of the monorepo, we shall use `storybook`

### PR DESCRIPTION
**Pull request by rannyeli**
Closes #6

### CODE REVIEW

**Comment by syncush** [comment]
Adding storybook-info is important, we can share the code snippets, @CptSchnitz  what do you think? 
reference for storybook info can be found [here](https://github.com/storybookjs/storybook/tree/master/addons/info)

**Review by syncush — CHANGES_REQUESTED** [review]
The smart table component does not have any items in it. we need to show the user the smart table in multiple states:
- Empty
- With a few rows (under the page limit)
- With multiple pages

**Review by syncush — CHANGES_REQUESTED** [review]
1. Add the following addons: `story`, `actions`, `knobs`, `info`.

2. Change folder structure such that each component has a stories folder and the storybook knows how to dynamically load all the stories.

3. Toggle `dark theme`.

File: mc-react-components/src/lib/date-range-picker/stories/DateRangePicker.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import { DateTimeRangePicker } from '../date-range-picker';
+
+export default {
+  title: 'Range Picker',
+  component: DateTimeRangePicker,
+};
+
+export const DateTime = () => <DateTimeRangePicker onChange={() => null}/>;
```
Comment (CptSchnitz) [inline mc-react-components/src/lib/date-range-picker/stories/DateRangePicker.stories.tsx]: You can add an action on every property that is an event/function:
```
export const DateTime = () => <DateTimeRangePicker onChange={action('date change')}/>;
```
leads to 
image

File: mc-react-components/src/lib/date-range-picker/stories/DateRangePicker.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+DateTime.story = {
+    name: 'Date time',
```
Comment (CptSchnitz) [inline mc-react-components/src/lib/date-range-picker/stories/DateRangePicker.stories.tsx]: date time range

File: mc-react-components/src/lib/map-filter-container/stories/Map.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+const mapDivStyle = {
+  "height": "100%",
+  "width": "100%",
+  "position": "fixed" as const,
+};
+export const Basic = () => <div style={mapDivStyle}>
+    <ContainerMap onPolygonSelection={() => null}/>
+  </div>;
+
+export const withDrawer = () => <MapFilterContainer handlePolygonReset={() => null} handlePolygonSelected={() => null}/>;
```
Comment (CptSchnitz) [inline mc-react-components/src/lib/map-filter-container/stories/Map.stories.tsx]: same about action

File: mc-react-components/src/lib/map-filter-container/stories/Select.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+};
+
+export const PolygonBox = () => {    
+    const [active, setActive] = useState(false);
+
+    return <PolygonSelectionUi 
+    isSelectionEnabled={active} 
+    onStartDraw={() => setActive(true)} 
+    onCancelDraw={() => setActive(false)} 
+    onReset={() => null}/>;
```
Comment (CptSchnitz) [inline mc-react-components/src/lib/map-filter-container/stories/Select.stories.tsx]: actions

File: mc-react-components/src/lib/smart-table/stories/SmartTable.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+import { button } from '@storybook/addon-knobs';
+
+export default {
+  title: 'Smart Table',
+  component: SmartTable,
+};
+
+interface Person {
+  first_name: string;
+  last_name: string;
```
Comment (CptSchnitz) [inline mc-react-components/src/lib/smart-table/stories/SmartTable.stories.tsx]: should be camelCase.

**Review by CptSchnitz — CHANGES_REQUESTED** [review]
About the dark mode:
you can try combining [this](https://github.com/hipstersmoothie/storybook-dark-mode#react) with 
https://github.com/MapColonies/conflicts-ui/blob/e92d83c609dfb2d33051ef93006a52d57af9260e/src/App.tsx#L11-L27
and see if it looks ok.

File: mc-react-components/src/lib/date-range-picker/stories/DateRangePicker.stories.tsx (hunk 1 above)
Comment (rannyeli) [reply mc-react-components/src/lib/date-range-picker/stories/DateRangePicker.stories.tsx]: done

File: mc-react-components/src/lib/date-range-picker/stories/DateRangePicker.stories.tsx (hunk 2 above)
Comment (rannyeli) [reply mc-react-components/src/lib/date-range-picker/stories/DateRangePicker.stories.tsx]: done

File: mc-react-components/src/lib/map-filter-container/stories/Map.stories.tsx (hunk 3 above)
Comment (rannyeli) [reply mc-react-components/src/lib/map-filter-container/stories/Map.stories.tsx]: done

File: mc-react-components/src/lib/map-filter-container/stories/Select.stories.tsx (hunk 4 above)
Comment (rannyeli) [reply mc-react-components/src/lib/map-filter-container/stories/Select.stories.tsx]: done

File: mc-react-components/src/lib/smart-table/stories/SmartTable.stories.tsx (hunk 5 above)
Comment (rannyeli) [reply mc-react-components/src/lib/smart-table/stories/SmartTable.stories.tsx]: done

**Comment by rannyeli** [comment]
there is still bugs with the docs, related issues : #16 #17

**Review by syncush — APPROVED** [review]
Great job!
We need to prioritize the bugs, currently, it might be enough to see the map with the drawer in the story and not in docs preview, if in the future programmers will complain about it, then we would bump its priority

**Review by CptSchnitz — APPROVED** [review]

**Pull request merged**